test-gen-plugin/src/main/java/com/dylibso/chicory/maven/JavaTestGen.java

```diff
@@ -437,13 +448,25 @@ private List<Expression> generateAssertThrows(Command cmd, String wasmFile) {
                                 + "assertThrows("
                                 + exceptionType
                                 + ".class, () -> "
-                                + generateModuleInstantiation(cmd, wasmFile, null, null, null)
+                                + generateModuleInstantiation(cmd, wasmFile, null, null)
                                 + ")");
 
+        method.getBody().get().addStatement(assertThrows);
         if (cmd.text() != null) {
-            return List.of(assertThrows, exceptionMessageMatch(cmd.text()));
-        } else {
-            return List.of(assertThrows);
+            method.getBody().get().addStatement(exceptionMessageMatch(cmd.text()));
+        }
+
+        if (excluded) {
+            method.addAnnotation(
+                    new SingleMemberAnnotationExpr(
+                            new Name("Disabled"),
+                            new StringLiteralExpr("Validation test excluded")));
+        } else if (cmd.moduleType() != null && cmd.moduleType().toLowerCase().contains("text")) {
```

Comment:
`toLowerCase` should always accept a locale (e.g. `Locale.ROOT`); otherwise, another way to do this would be `equalsIgnoreCase`.